File: backend/src/main/java/com/example/upload/domain/post/post/dto/PostListParamDto.java
package com.example.upload.domain.post.post.dto;

import com.example.upload.standard.search.SearchKeywordType;
import lombok.Builder;

@Builder
public record PostListParamDto(
        SearchKeywordType keywordType,
        String keyword,
        Boolean listed,
        Boolean published,
        Integer pageSize,
        Integer page

) {

    public int getPage() {
        return page == null ? 0 : page;
    }

    public int getPageSize() {
        return pageSize == null ? 10 : pageSize;
    }
}

File: backend/src/main/java/com/example/upload/domain/post/post/dto/PostListParamDto.kt
package com.example.upload.domain.post.post.dto

import com.example.upload.standard.search.SearchKeywordType

@JvmRecord
data class PostListParamDto(
    val keywordType: SearchKeywordType,
    val keyword: String,
    val listed: Boolean?,
    val published: Boolean?,
    val page: Int?,
    val pageSize: Int?

) {
    fun getPage(): Int {
        return page ?: 0
    }

    fun getPageSize(): Int {
        return pageSize ?: 10
    }
}